thirdparty/DiskANN/include/diskann/utils.h

```diff
@@ -768,6 +768,76 @@ namespace diskann {
     return max_norm;
   }
 
+  template<typename T>
+  void prepare_base_for_cosine(const std::string in_file,
+                                        const std::string out_file) {
+    LOG_KNOWHERE_DEBUG_
+        << "Pre-processing base file by normalizing";
+    std::ifstream in_reader(in_file.c_str(), std::ios::binary);
+    std::ofstream out_writer(out_file.c_str(), std::ios::binary);
+    _u64          npts, in_dims, out_dims;
+
+    _u32 npts32, dims32;
+    in_reader.read((char*) &npts32, sizeof(uint32_t));
+    in_reader.read((char*) &dims32, sizeof(uint32_t));
+
+    npts = npts32;
+    in_dims = dims32;
+    out_dims = in_dims;
+    _u32 outdims32 = (_u32) out_dims;
+
+    out_writer.write((char*) &npts32, sizeof(uint32_t));
+    out_writer.write((char*) &outdims32, sizeof(uint32_t));
+
+    size_t               BLOCK_SIZE = 100000;
```

Comment:
Can we align this number with buffer we use in other places during the building process to keep the memory usage steady

Reply:
It's aligned with line 718 above.

Reply:
Which number are you referring to